golang/btcspv/validate_spv.go

```diff
@@ -184,3 +184,32 @@ func ValidateHeaderChain(headers []byte) (sdk.Uint, error) {
 	}
 	return totalDifficulty, nil
 }
+
+func (s SPVProof) Validate() (bool, error) {
+	// Calculate the Tx ID and compare it to the one in SPVProof
+	txid := CalculateTxID(s.Version, s.Vin, s.Vout, s.Locktime)
```

Comment:
Add a line here: 
```
validHeader, err := s.ConfirmingHeader.Validate()
if err != nil {
  return false, err
}
```


Then add a Validate fuction to `BitcoinHeader` that checks:
1. `bytes.Equal(Hash256(b.Raw), b.HashLE)`
2. `b.Hash` is `b.HashLE` backwards
3. same for `b.MerkleRoot`
4. `bytes.Equal(b.MerkleRootLE, ExtractMerkleRootLE(b.Raw))`
5. `bytes.Equal(b.PrevHash, ExtractPrevBlockHashLE(b.Raw))`

Reply:
need to add the call here, so that we validate the header when we validate the proof object